tests/test_database_types.py

```diff
@@ -178,60 +242,135 @@ def expand_params(testcase_func, param_num, param):
+    # For showing assertEqual differences under a certain length.
+    maxDiff = 10000
+
     @parameterized.expand(type_pairs, name_func=expand_params)
     def test_types(self, source_db, target_db, source_type, target_type, type_category):
-        start = time.time()
-
         self.src_conn = src_conn = CONNS[source_db]
         self.dst_conn = dst_conn = CONNS[target_db]
 
         self.connections = [self.src_conn, self.dst_conn]
         sample_values = TYPE_SAMPLES[type_category]
 
-        src_table_path = src_conn.parse_table_name("src")
-        dst_table_path = dst_conn.parse_table_name("dst")
+        # We need to include the database name because of e.g. Presto which has
+        # Postgres as a backing catalog and shouldn't re-use those.
+        src_table_name = f"src_{source_db.__name__.lower()}_{parameterized.to_safe_name(source_type)}_{N_SAMPLES}"
+        src_table_path = src_conn.parse_table_name(src_table_name)
         src_table = src_conn.quote(".".join(src_table_path))
+
+        dst_table_name = f"dst_{target_db.__name__.lower()}_{parameterized.to_safe_name(target_type)}_{src_table_name}"
+        # Since we insert `src` to `dst`, the `dst` can be different for the
+        # same type. 😭
+        dst_table_path = dst_conn.parse_table_name(dst_table_name)
         dst_table = dst_conn.quote(".".join(dst_table_path))
 
+        # If you want clean house from e.g. changing fakers.
         src_conn.query(f"DROP TABLE IF EXISTS {src_table}", None)
-        src_conn.query(f"CREATE TABLE {src_table}(id int, col {source_type});", None)
-        _insert_to_table(src_conn, src_table, enumerate(sample_values, 1))
+        dst_conn.query(f"DROP TABLE IF EXISTS {dst_table}", None)
 
-        values_in_source = src_conn.query(f"SELECT id, col FROM {src_table}", list)
+        self.table = TableSegment(
+            self.src_conn, src_table_path, "id", None, ("col",), quote_columns=False
+        )
+        self.table2 = TableSegment(
+            self.dst_conn, dst_table_path, "id", None, ("col",), quote_columns=False
+        )
 
-        dst_conn.query(f"DROP TABLE IF EXISTS {dst_table}", None)
-        dst_conn.query(f"CREATE TABLE {dst_table}(id int, col {target_type});", None)
-        _insert_to_table(dst_conn, dst_table, values_in_source)
+        start = time.time()
+        already_seeded = self._create_table(src_conn, src_table, source_type)
+        if not already_seeded:
+            self._insert_to_table(
+                src_conn, src_table, source_type, enumerate(sample_values, 1)
+            )
+        insertion_source_duration = round(time.time() - start, 2)
 
-        self.table = TableSegment(self.src_conn, src_table_path, "id", None, ("col",), quote_columns=False)
-        self.table2 = TableSegment(self.dst_conn, dst_table_path, "id", None, ("col",), quote_columns=False)
+        start = time.time()
+        already_seeded = self._create_table(dst_conn, dst_table, target_type)
```

Comment:
We should be dropping the tables when we're done, no? So this exists only in case the tests fail.. And what if it's an old table with different values? Better to drop the tables and re-create everything.

Reply:
We keep the tables around on purpose for performance with large `N_SAMPLES`. Only if you change the fakers would you need to drop them.

Reply:
But that's exactly what's going to happen if I check-out some old revision on git. And I won't understand why the tests are behaving strangely.

Reply:
OK, that's fair. I'm thinking then I will add a benchmarking "mode" via an environment variable where you pass the sample numbers. We need to select a small subset of the tests for this either way. I'll comment it for now to get the numbers I need.